api/cloud/oc_cloud_manager.c

```diff
@@ -140,14 +140,24 @@ check_expires_in(int64_t expires_in)
   return expires_in > UINT16_MAX ? UINT16_MAX : (uint16_t)expires_in;
 }
 
+static oc_cloud_error_t
+_register_handler_check_data_error(oc_client_response_t *data)
+{
+  if (data->code >= OC_STATUS_SERVICE_UNAVAILABLE) {
+    return CLOUD_ERROR_CONNECT;
+  }
+  if (data->code >= OC_STATUS_BAD_REQUEST) {
+    return CLOUD_ERROR_RESPONSE;
+  }
+  return CLOUD_OK;
+}
+
 static int
 _register_handler(oc_cloud_context_t *ctx, oc_client_response_t *data)
 {
-  if (data->code >= OC_STATUS_SERVICE_UNAVAILABLE) {
-    cloud_set_last_error(ctx, CLOUD_ERROR_CONNECT);
-    goto error;
-  } else if (data->code >= OC_STATUS_BAD_REQUEST) {
-    cloud_set_last_error(ctx, CLOUD_ERROR_RESPONSE);
+  oc_cloud_error_t err = _register_handler_check_data_error(data);
+  if (err != CLOUD_OK) {
+    cloud_set_last_error(ctx, err);
```

Comment:
This here and `cloud_set_cps_and_last_error` in the `goto` label seems a bit redundant (though they set a different last error). Is guess we want to call `oc_notify_observers` twice...?

Reply:
pls remove this line - there will be only `goto error`

Reply:
and in error there will be `cloud_set_cps_and_last_error(ctx, OC_CPS_FAILED, err);`